addons/web/static/src/js/core/ajax.js

```diff
@@ -310,18 +310,29 @@ function get_file(options) {
         .load(function () {
             try {
                 if (options.error) {
-                    var body = this.contentDocument.body;
-                    var nodes = body.children.length === 0 ? body.childNodes : body.children;
                     var errorParams = {};
-
-                    try { // Case of a serialized Odoo Exception: It is Json Parsable
-                        var node = nodes[1] || nodes[0];
-                        errorParams = JSON.parse(node.textContent);
-                    } catch (e) { // Arbitrary uncaught python side exception
+                    var head = this.contentDocument.head;
+                    if (head.childNodes.length && head.childNodes[0].textContent === '504 Gateway Time-out') { // Check is request timeout(esp. if you set a low proxy_read_timeout in nginx)?
```

Comment:
That seems extremely and unnecessarily brittle.

@ged-odoo do you know if modern web APIs would allow us to get rid of the iframe hack to download files? Or if there's a clean/proper way to detect an error response of some sort (e.g. 504) aside from parsing the page content and hoping it has the right bits?

Reply:
@xmo-odoo no idea. Could not find much in a quick search. This may be interesting: https://stackoverflow.com/a/36082038

but it uses the fetch API, which is not supported by IE11.  Maybe we could use it anyway, and have a fallback if it does not work, with this code here

Reply:
@ged-odoo ah right I'd completely forgotten about the Fetch API. Might be a good idea to implement get_file on Fetch with an iframe fallback, that way we can cleanly handle issues in the fetch-based version and people using older browser have the less-useful but still-mostly-working version.

Reply:
mmm it looks like Fetch might not even be necessary (though would be nice for better readability): xhr was extended to support [form data](https://developer.mozilla.org/en-US/docs/Web/API/FormData) as well as [blob responses](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType). From there on we can just use download.js or lift the relevant bits from it to actually download the file.

Reply:
@xmo-odoo @ged-odoo What we have to do with this point? Final conclusion?

Reply:
@jat-odoo I've rebuilt the entire get_file function & handled this case in what I think is a cleaner way in #25731 using updated/modern browser APIs obviating the need for iframes. Assuming @ged-odoo is OK with merging that I'll just remove this change from here when merging it.

Reply:
:+1:, Thanx for informing.